tests/unit/test_types.py

```diff
@@ -1440,3 +1445,390 @@ def test_maptype_alias():
+    try:
+        type_string_to_type_object("struct<x int, y: string")
+        raise AssertionError(
+            "Expected ValueError for mismatched bracket or parse error"
+        )
+    except ValueError as ex:
+        assert (
+            "Missing closing" in str(ex)
+            or "Mismatched" in str(ex)
+            or "syntax" in str(ex).lower()
+        ), f"Expected bracket or parse syntax error, got: {ex}"
+
+
+def test_type_string_to_type_object_implicit_struct_simple():
+    """
+    Verify that a comma-separated list of 'name: type' fields parses as a StructType,
+    even without 'struct<...>'.
+    """
+    dt = type_string_to_type_object("a: int, b: string")
+    assert isinstance(dt, StructType), f"Expected StructType, got {dt}"
+    assert len(dt.fields) == 2, f"Expected 2 fields, got {len(dt.fields)}"
+
+    expected_field_a = StructField("a", IntegerType(), nullable=True)
+    expected_field_b = StructField("b", StringType(), nullable=True)
+
+    assert (
+        dt.fields[0] == expected_field_a
+    ), f"Expected {expected_field_a}, got {dt.fields[0]}"
+    assert (
+        dt.fields[1] == expected_field_b
+    ), f"Expected {expected_field_b}, got {dt.fields[1]}"
+
+
+def test_type_string_to_type_object_implicit_struct_single_field():
+    """
+    Even a single 'name: type' with no commas should parse to StructType
+    if your parser logic treats it as an implicit struct.
+    """
+    dt = type_string_to_type_object("c: decimal(10,2)")
+    assert isinstance(dt, StructType), f"Expected StructType, got {dt}"
+    assert len(dt.fields) == 1, f"Expected 1 field, got {len(dt.fields)}"
+
+    expected_field_c = StructField("c", DecimalType(10, 2), nullable=True)
+    assert (
+        dt.fields[0] == expected_field_c
+    ), f"Expected {expected_field_c}, got {dt.fields[0]}"
+
+
+def test_type_string_to_type_object_implicit_struct_nested():
+    """
+    Test an implicit struct with multiple fields,
+    including nested array/map types.
+    """
+    dt = type_string_to_type_object("arr: array<int>, kv: map<string, decimal(5,2)>")
+    assert isinstance(dt, StructType), f"Expected StructType, got {dt}"
+    assert len(dt.fields) == 2, f"Expected 2 fields, got {len(dt.fields)}"
+
+    expected_field_arr = StructField("arr", ArrayType(IntegerType()), nullable=True)
+    expected_field_kv = StructField(
+        "kv", MapType(StringType(), DecimalType(5, 2)), nullable=True
+    )
+
+    assert (
+        dt.fields[0] == expected_field_arr
+    ), f"Expected {expected_field_arr}, got {dt.fields[0]}"
+    assert (
+        dt.fields[1] == expected_field_kv
+    ), f"Expected {expected_field_kv}, got {dt.fields[1]}"
+
+
+def test_type_string_to_type_object_implicit_struct_with_spaces():
+    """
+    Test spacing variations. E.g. "  col1  :   int  ,  col2  :  map< string , decimal(5,2) >  ".
+    """
+    dt = type_string_to_type_object(
+        "  col1 :  int  ,  col2 :   map< string , decimal(5,2) > "
+    )
```

Comment:
```suggestion
    dt = type_string_to_type_object(
        "  col1 :  int  ,  col2 :   map< string , decimal( 5 , 2 ) > "
    )
```
can we add spacing here too